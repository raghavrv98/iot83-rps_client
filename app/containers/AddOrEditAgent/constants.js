/*
 *
 * AddOrEditAgent constants
 *
 */

export const CREATE_AGENT_REQUEST = "app/AddOrEditAgent/CREATE_AGENT_REQUEST";
export const CREATE_AGENT_SUCCESS = "app/AddOrEditAgent/CREATE_AGENT_SUCCESS";
export const CREATE_AGENT_FAILURE = "app/AddOrEditAgent/CREATE_AGENT_FAILURE";

export const GET_AGENT_DETAILS = "app/AddOrEditAgent/GET_AGENT_DETAILS";
export const GET_AGENT_DETAILS_SUCCESS = "app/AddOrEditAgent/GET_AGENT_DETAILS_SUCCESS";
export const GET_AGENT_DETAILS_FAILURE = "app/AddOrEditAgent/GET_AGENT_DETAILS_FAILURE";
