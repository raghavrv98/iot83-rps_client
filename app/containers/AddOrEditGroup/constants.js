/*
 *
 * AddOrEditGroup constants
 *
 */

export const ON_SUBMIT_REQUEST = "app/AddOrEditGroup/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_FAILURE";

export const GET_GROUP_DETAILS = "app/AddOrEditGroup/GET_GROUP_DETAILS";
export const GET_GROUP_DETAILS_SUCCESS = "app/AddOrEditGroup/GET_GROUP_DETAILS_SUCCESS";
export const GET_GROUP_DETAILS_FAILURE = "app/AddOrEditGroup/GET_GROUP_DETAILS_FAILURE";
