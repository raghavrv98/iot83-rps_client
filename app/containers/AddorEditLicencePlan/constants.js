/*
 *
 * AddorEditLicencePlan constants
 *
 */

export const DEFAULT_ACTION = "app/AddorEditLicencePlan/DEFAULT_ACTION";

export const ON_SUBMIT_REQUEST = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_FAILURE";

export const GET_PLANS_LIST = "app/AddorEditLicencePlan/GET_PLANS_LIST";
export const GET_PLANS_LIST_SUCCESS = "app/AddorEditLicencePlan/GET_PLANS_LIST_SUCCESS";
export const GET_PLANS_LIST_FAILURE = "app/AddorEditLicencePlan/GET_PLANS_LIST_FAILURE";