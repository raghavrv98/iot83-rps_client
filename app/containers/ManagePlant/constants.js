/*
 *
 * ManagePlant constants
 *
 */

export const DEFAULT_ACTION = "app/ManagePlant/DEFAULT_ACTION";

export const GET_PLANTLIST = "app/ManagePlant/GET_PLANTLIST";
export const GET_PLANTLIST_SUCCESS = "app/ManagePlant/GET_PLANTLIST_SUCCESS";
export const GET_PLANTLIST_FAILURE = "app/ManagePlant/GET_PLANTLIST_FAILURE";

export const DELETE_PLANT = "app/ManagePlant/DELETE_PLANT";
export const DELETE_PLANT_SUCCESS = "app/ManagePlant/DELETE_PLANT_SUCCESS";
export const DELETE_PLANT_FAILURE = "app/ManagePlant/DELETE_PLANT_FAILURE";