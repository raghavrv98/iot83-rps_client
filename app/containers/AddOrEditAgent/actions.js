/*
 *
 * AddOrEditAgent actions
 *
 */

import { CREATE_AGENT_REQUEST,GET_AGENT_DETAILS } from "./constants";

export function createAgent(data) {
  return {
    type: CREATE_AGENT_REQUEST,
    payload: data,
  };
}
export function getAgentDetails(id) {
  return {
    type: GET_AGENT_DETAILS,
    id
  };
}
