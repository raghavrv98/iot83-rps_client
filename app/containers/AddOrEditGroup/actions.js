/*
 *
 * AddOrEditGroup actions
 *
 */

import { ON_SUBMIT_REQUEST,GET_GROUP_DETAILS } from "./constants";

export function onSubmitHandler(payload,id) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload,
    id
  };
}

export function getGroupDetails(id) {
  return {
    type: GET_GROUP_DETAILS,
    id
  };
}
