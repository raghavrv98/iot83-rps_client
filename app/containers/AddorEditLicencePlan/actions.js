/*
 *
 * AddorEditLicencePlan actions
 *
 */

import { DEFAULT_ACTION, ON_SUBMIT_REQUEST, GET_PLANS_LIST } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function onSubmitHandler(payload) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload
  };
}

export function getPlansList(){
  return{
    type: GET_PLANS_LIST
  }
}