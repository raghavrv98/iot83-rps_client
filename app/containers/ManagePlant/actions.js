/*
 *
 * ManagePlant actions
 *
 */

import { DEFAULT_ACTION,GET_PLANTLIST, DELETE_PLANT } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function managePlantList() {
  return {
    type: GET_PLANTLIST
  };
}

export function deleteHandler(id) {
  return {
    type: DELETE_PLANT,
    id
  };
}