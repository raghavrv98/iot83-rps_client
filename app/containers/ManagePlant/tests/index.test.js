import React, { Component } from 'react';
import { shallow } from 'enzyme';
import history from "utils/history";
import { managePlantList, deleteHandler } from "../actions";
import { GET_PLANTLIST, DELETE_PLANT } from "../constants";
import { ManagePlant, mapDispatchToProps } from "../index";

const plantList = [
    {
      _id: "9bed0d5ca79011e9863a0242ac110016",
      createdAt: 1563257514466,
      name: "edfedeefdf",
      other: "fvcerdfedfv",
      overallGeometry: "dfe3dfedc",
      plantId: "f29b79c09f0911e986810242ac110011",
      processFluid: "efe3dcfed",
      updatedAt: 1563257547048
    }
  ];

describe('< ManagePlant />', () => {

    const match = {
        params: {
            id: ""
        }
    };

    it('Expect to have unit tests specified', () => {
        expect(true).toEqual(true);
    });

    it('match snapshots', () => {
        const wrap = shallow(<ManagePlant match={match} managePlantList={managePlantList} history={history}/>)
        expect(wrap).toMatchSnapshot()
    })

    it("test componentWillMount", async () => {
        const match = {
            params: {
                id: "demoId"
            }
        };
        const managePlantList = jest.fn();
        spyOn(ManagePlant.prototype, "componentWillMount").and.callThrough();
        shallow(
            <ManagePlant
                history={history}
                match={match}
                managePlantList={managePlantList}
            />
        );
        expect(ManagePlant.prototype.componentWillMount).toHaveBeenCalledTimes(1);
        expect(managePlantList).toHaveBeenCalledTimes(1);
    });

    it('should test component when recieve props', () => {
        spyOn(ManagePlant.prototype, 'componentWillReceiveProps').and.callThrough();
        const wrapper = shallow(<ManagePlant managePlantList={managePlantList} list={undefined} listError={undefined} deleteSuccess={undefined} deleteFailure={undefined} />);
        expect(wrapper).toBeDefined();
        wrapper.setState({ plantList });
        wrapper.setProps({ list: plantList, listError: "No Error Found.", deleteSuccess: "bb157d6738bb4886b77fd25635301010", deleteFailure: "No Error Found in delete." });
        expect(ManagePlant.prototype.componentWillReceiveProps).toHaveBeenCalledTimes(1);
    })

    // it("clicking add button trigger onClick event", async () => {
    //     let expectedCount = history.length + 1
    //     const wrapper = shallow(<ManagePlant managePlantList={managePlantList} history={history} />);
    //     let addButton = wrapper.find("button.btn.btn-create");
    //     expect(addButton.length).toBe(1);
    //     addButton.simulate('click');
    //     expect(history.length).toEqual(expectedCount);
    //     expect(history.location.pathname).toEqual("/addOrEditPlant");
    //   });

      it("Function testing on mapDispatchToProps", () => {
        const mockParams = {
          params: {
            id: "demo"
          }
        };
        const dispatch = jest.fn();
        mapDispatchToProps(dispatch).managePlantList();
        expect(dispatch.mock.calls[0][0]).toEqual({
          type: GET_PLANTLIST
        });
        mapDispatchToProps(dispatch).deleteHandler(
          mockParams.params.id
        );
        expect(dispatch.mock.calls[1][0]).toEqual({
          type: DELETE_PLANT,
          id: "demo"
        });
      });
});