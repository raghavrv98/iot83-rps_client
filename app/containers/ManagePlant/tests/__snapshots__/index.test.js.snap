// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< ManagePlant /> match snapshots 1`] = `
<div
  className="appContent"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      ManagePlant
    </title>
    <meta
      content="Description of ManagePlant"
      name="description"
    />
  </HelmetWrapper>
  <div
    className="pageBreadcrumb"
  >
    <div
      className="flex-item fx-b70"
    >
      <p>
        <FormattedMessage
          defaultMessage="Manage Plants"
          id="app.common.messages.managePlant"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </p>
      <h5>
        <FormattedMessage
          defaultMessage="All Available Plants"
          id="app.containers.ManagePlant.title"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
         
      </h5>
    </div>
    <div
      className="flex-item fx-b30 text-right align-items-center"
    />
  </div>
  <SkeletonLoader
    mode="fullView"
    skeleton="cardSkeleton"
  />
</div>
`;
